server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -0,0 +1,69 @@
+#include "irods/get_resource_info_for_operation.h"
+
+#include "irods/getRemoteZoneResc.h"
+#include "irods/irods_logger.hpp"
+#include "irods/irods_resource_backport.hpp"
+#include "irods/irods_resource_redirect.hpp"
+
+#include <nlohmann/json.hpp>
+#include <fmt/format.h>
+
+#include <array>
+#include <cstring> // for strdup
+#include <string>
+
+int rs_get_resource_info_for_operation(rsComm_t* _rsComm, dataObjInp_t* _dataObjInp, char** _out_info)
+{
+    using log_api = irods::experimental::log::api;
+    if (_rsComm == nullptr || _dataObjInp == nullptr || _out_info == nullptr) {
+        return SYS_INVALID_INPUT_PARAM;
+    }
+    rodsServerHost_t* rodsServerHost = nullptr;
+    const int remoteFlag = getAndConnRemoteZone(_rsComm, _dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+
+    if (REMOTE_HOST == remoteFlag) {
+        return rc_get_resource_info_for_operation(rodsServerHost->conn, _dataObjInp, _out_info);
+    }
+
+    const char* op_type = getValByKey(&_dataObjInp->condInput, GET_RESOURCE_INFO_OP_TYPE_KW);
+    if (op_type == nullptr) {
+        return SYS_INVALID_INPUT_PARAM;
```

Comment:
Should we put a message on the `rError` stack for this case? Might help the caller debug the issue should this arise.

Reply:
Sounds reasonable to me.

We need to log an error message in case the admin needs to figure out why something isn't working.

Reply:
logging now.

Reply:
Think this is done.